File: 1/fsharp/Solution.fs
module Solution
open System

let S1 (input : string) = 
    let split = input.Split [| '\n' |]
    let numbers = split |> Seq.map int
    let enumerator = numbers.GetEnumerator()

    enumerator.MoveNext() |> ignore
    let mutable lastValue = enumerator.Current
    let mutable count = 0

    while enumerator.MoveNext() do
        if enumerator.Current > lastValue then
            count <- count + 1
        lastValue <- enumerator.Current
    
    Console.WriteLine ("__S1__ = {0}", count)
    ()

let S2 (input : string) =
    let enumerator = (input.Split [| '\n' |] |> Seq.map int).GetEnumerator()
    
    let rec doThing prev lll ll l =
        let sum = lll + ll + l
        let res = if sum > prev then 1 else 0

        if enumerator.MoveNext() then
            res + (doThing sum ll l enumerator.Current)
        else
            res
    
    enumerator.MoveNext() |> ignore
    let lll = enumerator.Current
    enumerator.MoveNext() |> ignore
    let ll = enumerator.Current
    enumerator.MoveNext() |> ignore
    let l = enumerator.Current
    let sum = lll + ll + l

    enumerator.MoveNext() |> ignore
    let res = doThing sum ll l enumerator.Current

    Console.WriteLine("__S2__ = {0}", res)
    ()
    
let Solution input =
    S1 input
    S2 input
    ()

File: 1/csharp/Solution.cs
using System;
using System.Linq;

static class Solution
{
    public static void Run(string input)
    {
        S1(input);
        S2(input);
    }

    static void S1(string input)
    {
        int lastValue = -1;
        int count = 0;
        foreach(var num in input.Split("\n").Select(str => Convert.ToInt32(str)))
        {
            if(lastValue == -1)
            {
                lastValue = num;
                continue;
            }

            if(num > lastValue)
                count++;
            lastValue = num;
        }

        Console.WriteLine("__S1__ = {0}", count);
    }

    static void S2(string input)
    {
        var numbers = input.Split("\n").Select(str => Convert.ToInt32(str)).ToArray();
        int lastSum = -1;
        int count = 0;
        for(int i = 2; i < numbers.Length; i++)
        {
            int sum = numbers[i-2] + numbers[i-1] + numbers[i];
            if(lastSum != -1 && sum > lastSum)
                count++;
            lastSum = sum;
        }

        Console.WriteLine("__S2__ = {0}", count);
    }
}